test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1648,20 +1648,9 @@ private void restartNode(NodeAndClient nodeAndClient, RestartCallback callback)
         final Settings newSettings = nodeAndClient.closeForRestart(callback);
         removeExclusions(excludedNodeIds);
 
-        boolean success = false;
-        try {
-            nodeAndClient.recreateNode(newSettings, () -> rebuildUnicastHostFiles(emptyList()));
-            nodeAndClient.startNode();
-            success = true;
-        } finally {
-            if (success == false) {
-                removeNode(nodeAndClient);
-            }
-        }
-
-        if (activeDisruptionScheme != null) {
```

Comment:
why remove this? Is it a NOOP?

Reply:
We now call publishNode, which both adds to the nodes map and applies active disruption scheme.